alchemy/src/scope.ts

```diff
@@ -466,6 +477,8 @@ export class Scope {
     });
 
     if (!this.parent && process.env.ALCHEMY_TEST_KILL_ON_FINALIZE) {
```

Comment:
This only runs during our tests? Of are you also catching a signal and running it?